src/test/java/com/dreamsportslabs/guardian/it/RsaKeyGenerationIT.java

```diff
@@ -0,0 +1,358 @@
+package com.dreamsportslabs.guardian.it;
+
+import static com.dreamsportslabs.guardian.utils.ApplicationIoUtils.generateRsaKey;
+import static org.apache.commons.lang3.RandomStringUtils.randomAlphanumeric;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.not;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+
+import io.restassured.response.Response;
+import io.vertx.core.json.JsonObject;
+import java.security.KeyFactory;
+import java.security.interfaces.RSAPrivateKey;
+import java.security.interfaces.RSAPublicKey;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.security.spec.X509EncodedKeySpec;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Map;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+@Slf4j
+public class RsaKeyGenerationIT {
+
+  private static final String TENANT_VALID = "tenant1";
+  private static final String TENANT_UNKNOWN = randomAlphanumeric(8);
+
+  private JsonObject extractResponseBody(Response response) {
+    return new JsonObject(response.getBody().asString());
+  }
+
+  @Test
+  @DisplayName("Should generate RSA key with defaults successfully")
+  void generateRsaKeyWithDefaults() {
+    // Arrange
+    Map<String, Object> requestBody = new HashMap<>();
+
+    // Act
+    Response response = generateRsaKey(TENANT_VALID, requestBody);
+
+    // Assert
+    assertThat(response.getStatusCode(), equalTo(200));
+
+    JsonObject responseBody = extractResponseBody(response);
+
+    // Verify response structure
+    assertThat(responseBody.getString("kid"), notNullValue());
+    assertThat(responseBody.getValue("publicKey"), notNullValue());
+    assertThat(responseBody.getValue("privateKey"), notNullValue());
+    assertThat(responseBody.getInteger("keySize"), equalTo(2048));
+
+    // Verify PEM format (default)
+    String publicKey = responseBody.getString("publicKey");
+    String privateKey = responseBody.getString("privateKey");
+
+    assertThat(publicKey, containsString("-----BEGIN PUBLIC KEY-----"));
+    assertThat(publicKey, containsString("-----END PUBLIC KEY-----"));
+    assertThat(privateKey, containsString("-----BEGIN PRIVATE KEY-----"));
+    assertThat(privateKey, containsString("-----END PRIVATE KEY-----"));
+
+    // Verify key length (approximate check for PEM format)
+    assertThat(publicKey.length(), greaterThan(300));
+    assertThat(privateKey.length(), greaterThan(1500));
```

Comment:
We should use a key parsing lib to verify the key length and other details of the key.